python/ray/rllib/agent.py

```diff
@@ -82,12 +81,15 @@ def __init__(
             logdir_suffix = "{}_{}_{}".format(
                 env, self._agent_name,
                 datetime.today().strftime("%Y-%m-%d_%H-%M-%S"))
-            if not os.path.exists(self._default_logdir):
-                os.makedirs(self._default_logdir)
+            if not os.path.exists(DEFAULT_RESULTS_DIR):
+                os.makedirs(DEFAULT_RESULTS_DIR)
             self.logdir = tempfile.mkdtemp(
-                prefix=logdir_suffix, dir=self._default_logdir)
+                prefix=logdir_suffix, dir=DEFAULT_RESULTS_DIR)
             self._result_logger = UnifiedLogger(self.config, self.logdir, None)
 
+        # Also set the logdir as the pwd for user-defined loggers.
```

Comment:
This is not very visible - would it be reasonable to move this to docstring? Also the "for user-defined loggers" part is a little confusing - would be ok with removing it.

Reply:
On second thought, I realized chdir is actually a problem when running agents in the same process. I moved this code to the trial class, but I guess in general there's not a great solution here.

Reply:
this is fine; will merge after tests